test/qa/test_connect.py

```diff
@@ -450,3 +452,25 @@ def test_status_connected(tech, proto, obfuscated):
         assert time_passed - 1 <= time_connected <= time_passed + 1
 
     disconnect_base_test()
+
+
+@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.STANDARD_TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(40)
+def test_connect_to_virtual_server(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+    sh.nordvpn.set("virtual-location", "on")
```

Comment:
might be good to also have tests for virtual location `off`. That no virtual servers are returned and user cannot connect to one.

Reply:
There is a separate task to add tests related to settings, so I guess it covers this case.